### ISSUE TITLE
SyntaxWarning: invalid escape sequence \w

### ISSUE THREAD

**Issue by cjerdonek** [pyparsing/pyparsing#80]
Someone reported a `SyntaxWarning` when using pyparsing with Python 3.8: https://github.com/pypa/pip/issues/6362

It's caused by the following line starting with `make_html =` inside a docstring: https://github.com/pyparsing/pyparsing/blob/fed0f3da669a9a81829651080b6dfe32ce44c1b0/pyparsing.py#L3073-L3083

**Comment by pfmoore** [comment]
Note that the quoted line looks OK, but it's in a docstring that uses `"""` (not `r"""`), so backslashes are getting interpreted. The `\1` and `\2` on the same line are probably getting rendered wrongly as well...

Confirmed:
```
>>> help(pyparsing.Regex.sub)                                                                                       
Help on function sub in module pip._vendor.pyparsing:                                                               
                                                                                                                    
sub(self, repl)                                                                                                     
    Return Regex with an attached parse action to transform the parsed                                              
    result as if called using `re.sub(expr, repl, string) <https://docs.python.org/3/library/re.html#re.sub>`_.     
                                                                                                                    
    Example::                                                                                                       
                                                                                                                    
        make_html = Regex(r"(\w+):(.*?):").sub(r"<☺>☻</☺>")                                                         
        print(make_html.transformString("h1:main title:"))                                                          
        # prints "<h1>main title</h1>"                                                                              
```

**Comment by cjerdonek** [comment]
I edited my original post to include the surrounding lines so one can see that it's part of a docstring.

**Comment by ptmcg** [comment]
Thanks, this will be easily remedied before 3.8 is out. It looks like I'll need to get the 3.8a3 version and run some tests!

**Comment by ptmcg** [comment]
Included in release 2.4.0
